### ISSUE TITLE
Enforce/disallow `v-bind` same name shorthand style

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/eslint-plugin-vue#2359]
**Please describe what the rule should do:**

`v-bind` same name shorthand was [added in Vue 3.4](https://blog.vuejs.org/posts/vue-3-4#v-bind-same-name-shorthand) and is supported since #2355/#2357. A new rule (e.g. `vue/v-bind-same-name-style`) or a new option of the existing [`vue/v-bind-style`](https://eslint.vuejs.org/rules/v-bind-style.html) rule should be added that allows always enforcing that shorthand where possible, or always disallow it.

**What category should the rule belong to?**
[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

### `"sameNameShorthand": "always"`

```vue
<template>
  <!-- ✓ GOOD -->
  <div :foo />
  <div v-bind:foo />

  <div :foo="bar" />
  <div :foo='bar' />
  <div v-bind:foo="bar" />
  <div v-bind:foo='bar' />

  <!-- ✗ BAD -->
  <div :foo="foo" />
  <div :foo='foo' />
  <div v-bind:foo="foo" />
  <div v-bind:foo='foo' />
</template>
```

### `"sameNameShorthand": "never"`

```vue
<template>
  <!-- ✓ GOOD -->
  <div :foo="foo" />
  <div :foo='foo' />
  <div v-bind:foo="foo" />
  <div v-bind:foo='foo' />

  <div :foo="bar" />
  <div :foo='bar' />
  <div v-bind:foo="bar" />
  <div v-bind:foo='bar' />

  <!-- ✗ BAD -->
  <div :foo />
  <div v-bind:foo />
</template>
```

**Additional context:**

* #2355 
* #2357
* #2326

**Comment by waynzh** [comment]
I’d be happy to help out with this one!
And I've noticed that the AST structure generated by both ways seems to be the same, as per the [eslint-vue-parser](https://github.com/vuejs/vue-eslint-parser/blob/master/src/template/index.ts#L731). It looks like we can't differentiate them just by checking the `directive.value`.

Would love to get some insights on how we might tell them apart and get it working. Any thoughts?

**Comment by waynzh** [comment]
> We already have a function that checks if v-bind same-name shorthand.

So it's done through range comparison, that makes sense! Thanks for the reference.
